components/atoms/Select/custom-select.tsx

```diff
@@ -10,72 +13,68 @@ interface SelectProps {
-      }`}
-    >
-      <div
-        onClick={() => handleToggle()}
-        className="flex px-4 py-1.5 bg-white border border-light-slate-6 rounded-lg focus-within:border-light-orange-9 focus-within:ring focus-within:ring-light-orange-5 items-center"
-      >
-        {label && <span className="text-sm inline-flex text-light-slate-9 mr-2">{label}:</span>}
-        <input
-          value={selected.name === "" ? placeholder : selected.name}
-          readOnly
-          type="text"
-          className="overflow-scroll text-sm cursor-pointer text-light-slate-12 focus:outline-none bg-transparent"
-        />
-        <div className="w-6 h-4 relative overflow-hidden">
-          {" "}
-          <RiArrowUpSLine className="absolute bottom-1" /> <RiArrowDownSLine className="absolute top-1  " />
-        </div>
-      </div>
-      {isOpen && (
-        <div className="relative w-full">
-          <div className="left-0 right-0 transition bg-white overflow-hidden z-50 rounded-lg border shadow-superlative absolute font-normal ">
-            {options
-              ? options.map((option, index) => (
-                <div
-                  className="  cursor-pointer text-sm text-light-slate-12 hover:text-light-orange-11 hover:bg-light-orange-3 py-2 md:py-1 px-4 transition"
-                  onClick={() => handleSelected(option)}
-                  key={index}
-                >
-                  {option.name}
-                </div>
-              ))
-              : "No options"}
-          </div>
-        </div>
-      )}
+    <div className={clsx(
+      "radix-state-open:ring radix-state-open:ring-light-orange-5",
+      "relative cursor-pointer text-base items-center overflow-x-hidden",
+      "bg-white text-light-slate-10 rounded-lg min-w-max",
+      className || ""
+    )}>
+      <Select.Root onValueChange={handleSelected}>
+        <Select.Trigger
+          aria-label="Select a limit for the number of repositories to display"
+          className={clsx(
+            "radix-state-open:border-light-orange-9",
+            "inline-flex items-center gap-1 px-4 py-1.5",
+            "w-full text-sm outline-none rounded-lg font-semibold",
+            "text-light-slate-12 bg-white border border-light-slate-6")}
+        >
+          {label && <span className="inline-flex text-light-slate-9">{label}:</span>}
+          <Select.Value placeholder={placeholder} />
+          <Select.Icon className="w-4 h-4 relative overflow-hidden">
+            <RiArrowUpSLine className="absolute bottom-1" />
+            <RiArrowDownSLine className="absolute top-1" />
+          </Select.Icon>
+        </Select.Trigger>
+        <Select.Portal>
+          <Select.Content
+            className={clsx("absolute right-0 top-9 bg-white overflow-hidden rounded-lg border shadow-superlative", label ? "-left-18" : "left-0")}
```

Comment:
change the classname to `className={clsx("right-0 bg-white overflow-hidden rounded-lg border shadow-superlative")}` to keep the select menu working natively